### PR TITLE
Added Access Profiles and Roles with Import/Export CSV Functionality

### PR DESCRIPTION
**Pull request by richastral**
I have added the following nodes to the tenants:
- Access Profiles - which lists all access profiles, ability to creaate new, save and edit as well as import and export as CSV.
- Roles - which lists all roles, ability to create new, save and edit as well as import and export as CSV.

Mainly new classes, with minimal modification to existing code, where possible I have kept to your standards also.

Other changes:
- I have created a GenericCSVReader which is a clone of CSVReader but allows any objed to be read as CSV.
- I have made a modification to the CSVWriter to have an option transform parameter, this allows manipulation of the CSV record which is utilised for the the CSV import where I am listing entitlements in a single column of the CSV.
- I have added 2 new icons for import and export.

When creating new Access Profiles it prompts for Name, Owner, Source and an entitlement.  When creaing new Role it prompts for name, Owner and Access Profile.

Please review, this is functionality that we required for a client issue and has significanlty decreased time importing Access Profiles.

### CODE REVIEW

**Comment by yannick-beot-sp** [comment]
Thank you for this huge contribution. This looks awesome and I often get requests around access profiles and roles.
I will need quite some time to review this. Bear with me on this.

**Comment by richastral** [comment]
Thanks, I hope it can be of some use.

As I said I have made minimal changes to the original codebase, mainly class additions.

Let me know how you go.

**Comment by richastral** [comment]
Found an issue where there are more than 250 entitlements, it doesn't page, I have created a function to page the entitlements, works well, I just need to test fully before adding to the Pull Request.

Also will add a log file on the CSV Export so users can see why certain records have not been imported.

File: src/commands/access-profile/AccessProfilePaginator.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { AccessProfile } from "../../models/AccessProfile";
+import { IdentityNowClient } from "../../services/IdentityNowClient";
+
+/**
+ * Don't actually need this class, so will remove it in future as the getAccessProfiles()
```
Comment (yannick-beot-sp) [inline src/commands/access-profile/AccessProfilePaginator.ts:5]: The idea here is to "yield" results as they are coming.
So we write continuously the results, limiting the memory consumption.

File: src/services/IdentityNowClient.ts [hunk 2]
```diff
@@ -1110,6 +1110,215 @@ export class IdentityNowClient {
 			const path = `/beta/entitlements/${entitlementId}`;	
 			return await this.patchResource(path, JSON.stringify(payload));
 		}
+
+	/**
+	 * List Access Profiles
+	 * @returns list of access profiles
+	 * Added by richastral 06/07/2023
+	 */
+	public async getAccessProfiles(): Promise<any> {
```
Comment (yannick-beot-sp) [inline src/services/IdentityNowClient.ts:1119]: I have started to use the sailpoint-api-client (aka the TypeScript SDK).
So it limits the code I need to write AND the definition of data model/interfaces.
I'll take care of this

**Comment by yannick-beot-sp** [comment]
Closing this PR, as this is part of https://github.com/yannick-beot-sp/vscode-sailpoint-identitynow/pull/55 now

**Pull request closed without merging**